scripts/ccpp_database_obj.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+"""
+Define the CCPPDatabaseObj object
+Object definition and methods to provide information from a run of capgen.
+"""
+
+from host_model import HostModel
+from ccpp_suite import API
+
+class CCPPDatabaseObjError(ValueError):
+    """Error class specific to CCPPDatabaseObj.
+    All uses of this error should be internal (i.e., programmer error,
+       not user error)."""
+
+    def __init__(self, message):
+        """Initialize this exception"""
+        super().__init__(message)
+
+class CCPPDatabaseObj:
+    """Ojbect with data and methods to provide information from a run of capgen.
+    """
+
+    def __init__(self, run_env, host_model=None, api=None, database_file=None):
+        """Initialize this CCPPDatabaseObj.
+        If <database_file> is not None, all other inputs MUST be None and
+           the object is created from the database table created by capgen.
+        To initialize the object from an in-memory capgen run, ALL other
+           inputs MUST be passed (i.e., not None) and it is an error to pass
+           a value for <database_file>.
+        """
+
+        runtime_obj = all([host_model is not None, api is not None])
+        self.__host_model = None
+        self.__api = None
+        self.__database_file = None
+        if runtime_obj and database_file:
+            emsg = "Cannot provide both runtime arguments and database_file."
+        elif (not runtime_obj) and (not database_file):
+            emsg = "Must provide either database_file or all runtime arguments."
+        else:
+            emsg = ""
+        # end if
+        if emsg:
+            raise CCPPDatabaseObjError(f"ERROR: {emsg}")
+        # end if
+        if runtime_obj:
+            self.__host_model = host_model
+            self.__api = api
+        else:
+            self.db_from_file(run_env, database_file)
+        # end if
+
+    def db_from_file(self, run_env, database_file):
+        """Create the necessary internal data structures from a CCPP
+              datatable.xml file created by capgen.
+        """
+        metadata_tables = {}
+        host_name = "cam"
```

Comment:
I see that `host_name` is hardcoded to `cam`. What is the plan for making this configurable in the future?

Reply:
@peverwhee Can you comment on this please?

Reply:
@climbfuji Since the creating a database object from a file is not yet implemented, this host name is not currently doing anything. I did change it to something more general in the interim, but I think long-term, it will be parsed from the datafile.